tracing-opentelemetry/src/main/java/io/micronaut/tracing/opentelemetry/instrument/grpc/GrpcClientTracingInterceptorFactory.java

```diff
@@ -0,0 +1,50 @@
+/*
+ * Copyright 2017-2022 original authors
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+package io.micronaut.tracing.opentelemetry.instrument.grpc;
+
+import io.grpc.ClientInterceptor;
+import io.micronaut.context.annotation.Bean;
+import io.micronaut.context.annotation.Factory;
+import io.micronaut.context.annotation.Requires;
+import io.opentelemetry.api.OpenTelemetry;
+import io.opentelemetry.instrumentation.grpc.v1_6.GrpcTracing;
+import jakarta.inject.Singleton;
+
+import javax.annotation.Nonnull;
+
+/**
+ * Factory that builds the Tracing interceptors.
+ *
+ * @author Nemanja Mikic
+ */
+@Factory
+public class GrpcClientTracingInterceptorFactory {
```

Comment:
It's better to write a test, this can be an example: https://github.com/micronaut-projects/micronaut-grpc/blob/master/grpc-client-runtime/src/test/groovy/io/micronaut/grpc/server/tracing/GrpcTracingSpec.groovy